transport/server.cc

```diff
@@ -698,12 +698,21 @@ future<> cql_server::connection::process_request() {
             });
             auto istream = buf.get_istream();
             (void)_process_request_stage(this, istream, op, stream, seastar::ref(_client_state), tracing_requested, mem_permit)
-                    .then_wrapped([this, buf = std::move(buf), mem_permit, leave = std::move(leave)] (future<foreign_ptr<std::unique_ptr<cql_server::response>>> response_f) mutable {
+                    .then_wrapped([this, buf = std::move(buf), mem_permit, leave = std::move(leave), stream] (future<foreign_ptr<std::unique_ptr<cql_server::response>>> response_f) mutable {
                 try {
-                    write_response(response_f.get0(), std::move(mem_permit), _compression);
+                    if (response_f.failed()) {
```

Comment:
> But under what conditions can it happen? I see process_request_one() handles exceptions and returns a make_error() response. 

When that `process_request_one` exception handling logic throws, here's [the scenario](https://github.com/scylladb/scylladb/pull/11949/files#diff-98568d2760b9c091d956e50ca925da2a0f8d7562e3215b53b7876461678b9000R70) I ran into.

Reply:
Ok. Maybe we should move the error handling there? To keep everything related in one place.

Reply:
I'd like to leave it as is, since there is already a lot of code in `process_request_one`, adding another level of try/catch would complicate the things.